File: 5.Longest Palindromic substring/Solution.cpp
#include <iostream>
#include <string>

std::string longestPalindrome(std::string s) {
    int n = s.size();
    int start = 0, len = 0;

    for (int i = 0; i < 2 * n - 1; i++) {
        int left = i / 2, right = (i + 1) / 2;
        while (left >= 0 && right < n && s[left] == s[right]) {
            left--;
            right++;
        }
        if (right - left - 1 > len) {
            start = left + 1;
            len = right - left - 1;
        }
    }
    
    return s.substr(start, len);
}

int main() {

    return 0;
}

File: 5.Longest Palindromic substring/Solution.ts
function longestPalindrome(s: string): string{
    const n = s.length
    let start = 0, end = 0

    for(let i = 0; i < 2 * n - 1; i++) {
        let left = Math.floor(i / 2), right = Math.floor((i + 1) / 2)
        while (left >= 0 && right < n && s[left] === s[right]) {
            left--
            right++
        }
        if (right - left > end - start) {
            end = right
            start = left
        }
    }
    return s.substring(start + 1, end)
}